bftengine/include/bftengine/DbCheckpointManager.hpp

```diff
@@ -64,11 +64,19 @@ class DbCheckpointManager {
    ***************************************************/
   std::optional<DbCheckpointMetadata::DbCheckPointDescriptor> getLastCreatedDbCheckpointMetadata();
 
+  // A callback that is called after creating a checkpoint. Its purpose is to prepare it for use. It receives as
+  // parameters:
+  //  * the block ID at which the checkpoint was created (it might be less than the last block ID in the checkpoint
+  //  itself)
+  //  * the path to the checkpoint
+  using PrepareCheckpointCallback = std::function<void(BlockId, const std::string&)>;
+
   Seconds getLastCheckpointCreationTime() const { return lastCheckpointCreationTime_; }
   void initializeDbCheckpointManager(std::shared_ptr<concord::storage::IDBClient> dbClient,
                                      std::shared_ptr<bftEngine::impl::PersistentStorage> p,
                                      std::shared_ptr<concordMetrics::Aggregator> aggregator,
-                                     const std::function<BlockId()>& getLastBlockIdCb);
+                                     const std::function<BlockId()>& getLastBlockIdCb,
```

Comment:
Passing lambda is a good idea, but for the same class, a private member function is preferable. Lambda for lambda calculus is better. If we have a thread that will use the lambda and execute later or if we require lambda to build concepts (c++20) then putting lambda is good. I am not saying to remove the lambda,  instead to see if we will require this function only once or whether this function will be required to be used at multiple places. If there is a chance to use the function at multiple places, then we can move the same as a private function.

Reply:
The idea behind the callback is that the code inside of it requires access to the `kvbc` library. Yet, we don't want to make the `corebft` library depend on `kvbc`. Using a callback allows us to remove this dependency.